core/procedures/observable_parameter_model.ts

```diff
@@ -48,6 +48,20 @@ export class ObservableParameterModel implements IParameterModel {
         'implement your own custom ParameterModel.');
   }
 
+  /**
+   * Returns the name of this parameter.
+   */
+  getName(): string {
+    return this.variable.name;
+  }
+
+  /**
+   * Returns the types of this parameter.
+   */
+  getTypes(): string[] {
+    return [];
```

Comment:
We're not supporting types in parameters either right? if so then I think it's mildly confusing that in the procedure it throws but here it returns empty. I think we should be consistent unless there is an outside reason this has to return empty.

also same comment applies about documenting the lack of support in this implementation

Reply:
In both classes the setter throws but the getter returns a value, so I think it is consistent in that way. But I feel you about how it's weird.

I built it this way so that the procedure serializer could just work against the interfaces, rather than the concrete implementations. That way if people have to implement their own procedure data models they don't have to also implement their own serializer. But the utility of that is questionable.

I /prefer/ allowing the built-in serializer to work against the abstract types rather than the concrete ones because it's more future proof. But I think I'm fine with going the other way on this as well.

Reply:
ah no that makes sense, i was just looking at the setter for one and getter for another. not forcing people to write their own serializer seems useful. cool beans!

Reply:
❄️🫘